### PR TITLE
Merge power op (positive fractional and integral exponent) in same signature on tt_lib.tensor

### PR DESCRIPTION
**Pull request by umadevimcw**
Power op: pow(x,exponent) is created to support exponent when it is a postiive integer or positive floating point value.
 - this merges the two alternative ops present earlier

### CODE REVIEW

**Review by muthutt — COMMENTED** [review]
its really nice to have the single pow Python API expose to user via tt_dnn and hide implementation details

File: tt_eager/tt_dnn/op_library/composite/composite_ops.hpp [hunk 1]
```diff
@@ -294,11 +294,14 @@ Tensor tril(const Tensor& input_a, int32_t diag = 0, const MemoryConfig& output_
 Tensor triu(const Tensor& input_a, int32_t diag = 0, const MemoryConfig& output_mem_config = operation::DEFAULT_OUTPUT_MEMORY_CONFIG);
 
 //power_fp : power with floating point exponent
-Tensor power_fp(const Tensor& input_a, float exponent, const MemoryConfig& output_mem_config = operation::DEFAULT_OUTPUT_MEMORY_CONFIG);
+//Tensor power_fp(const Tensor& input_a, float exponent, const MemoryConfig& output_mem_config = operation::DEFAULT_OUTPUT_MEMORY_CONFIG);
 
 //repeat a input tensor @input_a as specified by the number of dimensions
 Tensor repeat(const Tensor& input_a, const Shape& shape,const MemoryConfig& output_mem_config = operation::DEFAULT_OUTPUT_MEMORY_CONFIG);
 
+template <typename T>
```
[the remark this answers is not in the document]
Comment (umadevimcw) [reply tt_eager/tt_dnn/op_library/composite/composite_ops.hpp:302]: Done and updated the test as well

**Review by muthutt — COMMENTED** [review]
- after your PR passes in dev testing please fix formatting and make one commit (rebase)
- request reviews after moving from draft PR
- please remove commented codes introduced during dev

**Comment by umadevimcw** [comment]
@muthutt CI are passing except perf (perf failed with timed out issue)
Please find the CI link https://github.com/tenstorrent-metal/tt-metal/actions/runs/7470933951/job/20333465485

File: tests/tt_eager/python_api_testing/sweep_tests/test_configs/ci_sweep_tests/broken_grayskull/pytorch_eltwise_power_fp_test.yaml [hunk 2]
```diff
@@ -9,7 +9,7 @@ test-list:
         num-samples: 64
         args-sampling-strategy: "all"
       env:
-        TT_METAL_SLOW_DISPATCH_MODE: ["", "1"]
+        TT_METAL_SLOW_DISPATCH_MODE: [""]
```
Comment (tt-aho) [inline tests/tt_eager/python_api_testing/sweep_tests/test_configs/ci_sweep_tests/broken_grayskull/pytorch_eltwise_power_fp_test.yaml:12]: What's the reason these sweep tests are being modified regarding slow vs fast dispatch testing?

File: tests/tt_eager/python_api_testing/sweep_tests/test_configs/ci_sweep_tests/broken_grayskull/pytorch_eltwise_power_fp_test.yaml [hunk 3]
```diff
@@ -26,7 +26,7 @@ test-list:
         buffer-type: ["DRAM", "L1", "SYSTEM_MEMORY"]
         out-buffer-type: ["DRAM"]
       output-file: eltwise_power_fp_sweep.csv
```
Comment (TT-BrianLiu) [inline tests/tt_eager/python_api_testing/sweep_tests/test_configs/ci_sweep_tests/broken_grayskull/pytorch_eltwise_power_fp_test.yaml:28]: Change to `eltwise_pow_sweep.csv` to match op name changes elsewhere? Also, update the file names from `*_eltwise_power_fp_*` to `*_eltwise_pow__*`

**Review by TT-BrianLiu — CHANGES_REQUESTED** [review]

**Review by arakhmati — APPROVED** [review]

**Review by muthutt — APPROVED** [review]
LGTM - correspondingly we also need to update pow_bw in future

**Review by mywoodstock — APPROVED** [review]

File: tt_eager/tt_dnn/op_library/composite/composite_ops.cpp [hunk 4]
```diff
@@ -1261,23 +1260,34 @@ Tensor _power_fp(const Tensor& input_a, float exponent, const MemoryConfig& outp
...
     Tensor  pow_trunc_log = mul_unary( log(input_a,output_mem_config),
                                        exponent_trunc, output_mem_config );
     Tensor pow_frac = exp( pow_trunc_log, output_mem_config );
     pow_trunc_log.deallocate();
-
-    return mul( power(input_a, exponent_floor, output_mem_config),
+    Tensor  t_nan  = full_like(input_a, std::nanf(""), output_mem_config);
+    Tensor result =  mul( power(input_a, exponent_floor, output_mem_config),
 		pow_frac, {}, output_mem_config );
+    // To handle negative inputs:
```
[the remark this answers is not in the document]
Comment (umadevimcw) [reply tt_eager/tt_dnn/op_library/composite/composite_ops.cpp:1273]: Thanks

File: tests/tt_eager/python_api_testing/sweep_tests/test_configs/ci_sweep_tests/broken_grayskull/pytorch_eltwise_power_fp_test.yaml (hunk 3 above)
Comment (umadevimcw) [reply tests/tt_eager/python_api_testing/sweep_tests/test_configs/ci_sweep_tests/broken_grayskull/pytorch_eltwise_power_fp_test.yaml:28]: @TT-BrianLiu I have updated the sweep tests output file name. 
In order to differentiate between int exponent and float exponent tests i haven't renamed the .yaml files is that fine?

File: tests/tt_eager/python_api_testing/sweep_tests/test_configs/ci_sweep_tests/broken_grayskull/pytorch_eltwise_power_fp_test.yaml (hunk 2 above)
Comment (umadevimcw) [reply tests/tt_eager/python_api_testing/sweep_tests/test_configs/ci_sweep_tests/broken_grayskull/pytorch_eltwise_power_fp_test.yaml:12]: @tt-aho In WHB0 test is aborted if fast dispatch is enabled, hence the fast dispatch setting is disabled

Comment (tt-aho) [reply tests/tt_eager/python_api_testing/sweep_tests/test_configs/ci_sweep_tests/broken_grayskull/pytorch_eltwise_power_fp_test.yaml:12]: This file is for gs (broken_grayskull dir)

Comment (tt-aho) [reply tests/tt_eager/python_api_testing/sweep_tests/test_configs/ci_sweep_tests/broken_grayskull/pytorch_eltwise_power_fp_test.yaml:12]: Similarly, the one in broken_wormhole is expected to fail right? So should that have been modified to skip FD test?

Comment (umadevimcw) [reply tests/tt_eager/python_api_testing/sweep_tests/test_configs/ci_sweep_tests/broken_grayskull/pytorch_eltwise_power_fp_test.yaml:12]: @tt-aho Yes GS test is modified to test only FD and WHB0 test modified to test SD alone

File: tests/tt_eager/python_api_testing/sweep_tests/test_configs/ci_sweep_tests/broken_grayskull/pytorch_eltwise_power_fp_test.yaml (hunk 3 above)
Comment (TT-BrianLiu) [reply tests/tt_eager/python_api_testing/sweep_tests/test_configs/ci_sweep_tests/broken_grayskull/pytorch_eltwise_power_fp_test.yaml]: is it difficult to rename the yaml as well?

**Review by TT-BrianLiu — APPROVED** [review]

**Pull request merged**